cukinia

```diff
@@ -1111,7 +1128,7 @@ _junitxml_add_case()
 	__suite_tests=$((__suite_tests + 1))
 
 	cat >>"$logfile" <<EOF
-    <testcase classname="${__log_class:-cukinia}" name="$name" time="0">
+    <testcase classname="${__log_class:-cukinia}" name="$name" ${id:+id=\"${id}\"} time="0">
```

Comment:
I think the issue here is that `id=` is not a valid tag in the JUnit XML spec, any way to keep it compliant ?

Reply:
I don't know if there is really a common specification. Some are defining the id like here: https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=formats-junit-xml-format
some are not like here: https://www.checkov.io/8.Outputs/JUnit%20XML.html
Do you have the valid JUnit XML spec ?

Reply:
I found this one. Rationale being my comment is to avoid producing JUnitXML that either cannot be understood by the applications that consume it, or breaks validation if said app validates the schema before parsing.

``` xml
<?xml version="1.0" encoding="utf-8"?>
<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
  <xsl:output method="text" indent="no"/>

  <xsl:template match="/testsuite">
    Testsuite: <xsl:value-of select="@name" />
    <xsl:text>
    Tests run: </xsl:text>
    <xsl:value-of select="@tests" />
    <xsl:text>, Failures: </xsl:text>
    <xsl:value-of select="@failures" />
    <xsl:text>, Errors: </xsl:text>
    <xsl:value-of select="@errors" />
    <xsl:text>, Time elapsed: </xsl:text>
    <xsl:value-of select="@time" />
    <xsl:text> sec</xsl:text>
    <xsl:apply-templates select="system-out" />
    <xsl:apply-templates select="system-err" />
    <xsl:text>
    --------- ----------- ---------
    </xsl:text>
    <xsl:apply-templates select="testcase" />
  </xsl:template>

  <xsl:template match="testcase">
    <xsl:text>
    Testcase: </xsl:text>
    <xsl:value-of select="@name" />
    <xsl:text> took </xsl:text>
    <xsl:value-of select="@time" />
  </xsl:template>

  <xsl:template match="system-out">
    <xsl:text>
    ------ Standard output ------
    </xsl:text>
    <xsl:value-of select="." />
  </xsl:template>

  <xsl:template match="system-err">
    <xsl:text>
    ------ Error output ------
    </xsl:text>
    <xsl:value-of select="." />
  </xsl:template>

</xsl:stylesheet>
```

Reply:
Update: more recent version at https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd

Reply:
Then we have no other choice but to put it in "name" of testcase :/

Reply:
If I understand well the jenkins parser, there is the support of [https://github.com/jenkinsci/junit-plugin/blob/master/src/main/java/hudson/tasks/junit/CaseResult.java#L101](stdout and stderr)
And there is also a [https://github.com/jenkinsci/junit-plugin/blob/master/src/main/java/hudson/tasks/junit/CaseResult.java#L87](properties) fields with name and value that could be used for the test_id

Reply:
nice, looking at the example in [junit-report-with-properties.xml](https://github.com/jenkinsci/junit-plugin/blob/37061da1199fafe91232d6995e9f3c16ac8b7fbc/src/test/resources/hudson/tasks/junit/junit-report-with-properties.xml) it seems we can have per-testsuite and per-testcase properties.

Looking at the semantics for properties.name, something with only lowercase words and dots ( would match better than underscore (current name examples: `osgi.dev`, `java.class.path`, `eclipse.application`, `prg`, `user.dir`, `autotest.debug`,  etc.). Considering there is no standard and names are scoped per domain, `cukinia.id` would probably follow this logic. 


``` xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<testsuites>
  <testsuite name="io.jenkins.example.with.properties" time="2.000" tests="2" errors="0" skipped="0" failures="1">
    <properties>
      <property name="prop1" value="value1" />
      <property name="multiline">
          Config line 1
          Config line 2
          Config line 3
      </property>
    </properties>
    <testcase name="testCaseA" classname="io.jenkins.example.with.properties" time="1.000">
      <failure message="expected: [a] but was: [b]" type="org.junit.ComparisonFailure">
        org.junit.ComparisonFailure: expected: [a] but was: [b]
      </failure>
      <properties>
        <property name="description">description of test testCaseA</property>
      </properties>
      ```

Reply:
Implemented.
I took advantage of this xml work to resolve a bug on stdout and  stderr outputs and skip information